### PR TITLE
DOC: add feature docs for TCP health probe

### PR DESCRIPTION
**Pull request by stijnmoreels**
Adds feature docs for the TCP health probe.

Relates to #11

Ps.: the folder structure will probably change when more features are added; so I added this feature docs at the root of the `./features` folder.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ Coming soon!
 
 # Features
 
-Coming soon!
+- [TCP health probe](features/tcp-health-probe) allow the runtime to periodically check liveness/readiness of the host.
```
Comment (tomkerkhove) [inline docs/index.md]: How about this?
```suggestion
- Support for exposing TCP health probes to periodically check liveness/readiness of the host ([docs](features/tcp-health-probe))
```

File: docs/features/tcp-health-probe.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+## Usage
+
+To include the TCP endpoint, add the following line of code in the `Startup.ConfigureServices` method:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddTcpHealthProbes(healthBuilder => 
+    {
+        healthBuilder.AddCheck("Example", () => HealthCheckResult.Healthy("Example is OK!"), tags: new[] { "example" })
```
Comment (tomkerkhove) [inline docs/features/tcp-health-probe.md]: Maybe mention that this is optional and you can just do `services.AddTcpHealthProbes()`?

File: docs/features/tcp-health-probe.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
+---
+title: "TCP Health probe"
+layout: default
+---
+
+# TCP Health probe
+
+A `BackgroundService` can be added to the <span>ASP.NET</span> hosted services to expose a TCP health endpoint that allows a runtime to periodically check the liveness/readiness of the host.
```
Comment (tomkerkhove) [inline docs/features/tcp-health-probe.md]: I'm not sure if we have to go so technical, isn't it just enough to say that we provide a TCP health probe endpoint which will expose health information based on .NET Core health approach?

_PS: It's not ASP.NET here so would leave that out maybe?_

File: docs/index.md (hunk 1 above)
Comment (stijnmoreels) [reply docs/index.md]: Ah, yes; was thinking about the approach used in the other repo's.

File: docs/features/tcp-health-probe.md (hunk 3 above)
Comment (stijnmoreels) [reply docs/features/tcp-health-probe.md]: Ah, right; I'll change this.

File: docs/features/tcp-health-probe.md (hunk 2 above)
Comment (stijnmoreels) [reply docs/features/tcp-health-probe.md]: Yep, added this in the inline code now, with comments explaining.

File: docs/features/tcp-health-probe.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
+---
+title: "TCP Health probe"
+layout: default
+---
+
+# TCP Health probe
+
+We provide a TCP health probe endpoint that allows a runtime to periodically check the liveness/readiness of the host based on the .NET Core health approach.
```
Comment (tomkerkhove) [inline docs/features/tcp-health-probe.md]:
```suggestion
We provide a TCP health probe endpoint that allows a runtime to periodically check the liveness/readiness of the host based on the [.NET Core health checks](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/health-checks).
```

Comment (stijnmoreels) [reply docs/features/tcp-health-probe.md]: Ah, yes. Thanks!

**Pull request merged**